packages/video_player/video_player_avfoundation/darwin/Classes/FVPVideoPlayerPlugin.m

```diff
@@ -308,9 +311,20 @@ - (instancetype)initWithPlayerItem:(AVPlayerItem *)item
   // video streams (not just iOS 16).  (https://github.com/flutter/flutter/issues/109116). An
   // invisible AVPlayerLayer is used to overwrite the protection of pixel buffers in those streams
   // for issue #1, and restore the correct width and height for issue #2.
+  // It is also used to start picture-in-picture
   _playerLayer = [AVPlayerLayer playerLayerWithPlayer:_player];
+  // picture-in-picture shows a placeholder where the original video was playing.
+  // This is a native overlay that does not scroll with the rest of the Flutter UI.
+  // That is why we need to set the opacity of the overlay.
+  // Setting it to 0 would result in the picture-in-picture not working.
+  // Setting it to 1 would result in the picture-in-picture overlay always showing over other
+  // widget. Setting it to 0.001 makes the placeholder invisible, but still allows the
+  // picture-in-picture.
+  _playerLayer.opacity = 0.001;
   [self.flutterViewLayer addSublayer:_playerLayer];
 
+  [self setupPiPController];
```

Comment:
This violates the style guide; see https://google.github.io/styleguide/objcguide.html#avoid-messaging-the-current-object-within-initializers-and--dealloc (some existing code also does this, but that code is also incorrect, and we shouldn't add more of it).

You can instead make a static C function with the logic, which accepts `self` and sets values using `self->_foo = bar` instead of calling setters.